### PR TITLE
Revert "[SPARK-41677][CORE][SQL][SS][UI] Add Protobuf serializer for StreamingQueryProgressWrapper"

### PR DESCRIPTION
**Pull request by gengliangwang**
### What changes were proposed in this pull request?

This reverts commit 915e9c67a9581a1f66e70321879092d854c9fb3b.

### Why are the changes needed?

When running end-to-end tests, there are 5 NPE errors from string fields:

- SourceProgress.latestOffset
- SourceProgress.endOffset
- SourceProgress.startOffset
- StreamingQueryData.name
- StreamingQueryProgress.name

After fixing them, there is following error:
```
java.lang.UnsupportedOperationException
	at java.base/java.util.Collections$UnmodifiableMap.remove(Collections.java:1460)
	at org.apache.spark.sql.streaming.ui.StreamingQueryStatisticsPage.$anonfun$generateStatTable$27(StreamingQueryStatisticsPage.scala:401)
```
The deserialized map `StreamingQueryProgress.durationMs` needs to be mutable.

Give the StreamingQueryProgressWrapper contains nullable fields and mutable map, I suggest using the default JSON serailizer for this class.

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

GA tests

### CODE REVIEW

**Comment by gengliangwang** [comment]
cc @LuciferYang @techaddict @panbingkun

**Comment by gengliangwang** [comment]
I am running end-to-end tests from https://github.com/apache/spark/pull/39415

**Review by dongjoon-hyun — APPROVED** [review]
This is a clean revert, right?

**Comment by gengliangwang** [comment]
@dongjoon-hyun yes it is

File: sql/core/src/main/scala/org/apache/spark/status/protobuf/sql/StreamingQueryProgressSerializer.scala [hunk 1]
```diff
@@ -1,89 +0,0 @@
...
-
-  def deserialize(process: StoreTypes.StreamingQueryProgress): StreamingQueryProgress = {
-    new StreamingQueryProgress(
-      id = UUID.fromString(process.getId),
-      runId = UUID.fromString(process.getRunId),
-      name = process.getName,
-      timestamp = process.getTimestamp,
-      batchId = process.getBatchId,
-      batchDuration = process.getBatchDuration,
-      durationMs = process.getDurationMsMap,
```
[the remark this answers is not in the document]
Comment (gengliangwang) [reply sql/core/src/main/scala/org/apache/spark/status/protobuf/sql/StreamingQueryProgressSerializer.scala:70]: There are 4 `ju.map` in the whole object..and for the string fields some of them are nullable...
I feel that it is risky for this one.

[the remark this answers is not in the document]
Comment (gengliangwang) [reply sql/core/src/main/scala/org/apache/spark/status/protobuf/sql/StreamingQueryProgressSerializer.scala:70]: @LuciferYang If you are confident about this one. Feel free to add the serializer again after the tests are merged

[the remark this answers is not in the document]
Comment (LuciferYang) [reply sql/core/src/main/scala/org/apache/spark/status/protobuf/sql/StreamingQueryProgressSerializer.scala:70]: OK, I am fine to revert this one first

**Review by LuciferYang — APPROVED** [review]
+1, LGTM

File: sql/core/src/main/scala/org/apache/spark/status/protobuf/sql/StreamingQueryProgressSerializer.scala (hunk 1 above)
[the remark this answers is not in the document]
Comment (LuciferYang) [reply sql/core/src/main/scala/org/apache/spark/status/protobuf/sql/StreamingQueryProgressSerializer.scala:70]: Yes, there are risks. I think it is better to use default JSON serializer for this class for Spark 3.4.0

**Comment by gengliangwang** [comment]
@dongjoon-hyun @LuciferYang Thanks for the review.
Merging to master

**Pull request closed without merging**